tests/python_package_test/test_basic.py

```diff
@@ -609,3 +611,57 @@ def test_custom_objective_safety():
     good_bst_multi.update(fobj=_good_gradients)
     with pytest.raises(ValueError, match=re.escape(f"number of models per one iteration ({nclass})")):
         bad_bst_multi.update(fobj=_bad_gradients)
+
+@pytest.fixture(name="fake_model")
+def _fake_model() -> lgb.Booster:
+    # TODO: maybe removed deps on data_dir later
+    data_dir = Path(__file__).parent.parent.parent / "examples/binary_classification"
+
+    df_train = pd.read_csv(data_dir / "binary.train", header=None, sep="\t")
+    weights = pd.read_csv(data_dir / "binary.train.weight", header=None)[0]
+
+    X_train = df_train.drop(0, axis=1)
+    y_train = df_train[0]
+
+    params = {
+        "boosting_type": "gbdt",
+        "objective": "binary",
+        "metric": "binary_logloss",
+        "num_leaves": 31,
+        "learning_rate": 0.05,
+        "feature_fraction": 0.9,
+        "bagging_fraction": 0.8,
+        "bagging_freq": 5,
+        "verbose": 0,
+    }
+    lgb_train = lgb.Dataset(X_train, y_train, weight=weights, free_raw_data=False)
+    feature_name = [f"feature_{col}" for col in X_train.columns]
+
+    gbm = lgb.train(
+        params,
+        lgb_train,
+        num_boost_round=3,
+        valid_sets=lgb_train,  # eval training data
+        feature_name=feature_name,
+        categorical_feature=[21],
+    )
+
+    return gbm
+
+def test_booster_load_params_when_passed_model_file(fake_model: lgb.Booster) -> None:
```

Comment:
```suggestion
def test_booster_load_params_when_passed_model_file(fake_model):
```

Thank you very much for adding type hints! But they are not necessary for test functions (anything in a `tests/` files that starts `def test_*`).

Please remove these, for consistency with the rest of the tests.

Reply:
Why is test necessary in code but not in test?

Reply:
Thanks for the detailed explanation! My thought is just that with type hints, the code is much easier to understand in general. But I also prefer not to deviate from PR. I'll leave type hints out of test for now.